system-addon/lib/LinksCache.jsm

```diff
@@ -5,7 +5,10 @@
 
 this.EXPORTED_SYMBOLS = ["LinksCache"];
 
-const EXPIRATION_TIME = 5 * 60 * 1000; // 5 minutes
+// This should be slightly less than SYSTEM_TICK_INTERVAL as timers are
+// not precise enough causing skips in the refresh interval.
```

Comment:
nit: It's because timer comparisons are *too* exact ;) and the underlying issue is that async/await functionality will make the last recorded times a little bit later. Might be useful to include the 10%, e.g., "slightly less (10%)"